packages/dynamic_layouts/README.md

```diff
@@ -13,25 +13,135 @@ and the Flutter guide for
+`crossAxisSpacing`.
+
+The possibility to only have one of the axis be a decided by the children is possible by
+changing the values of `childCrossAxisExtent` and `childMainAxisExtent`. This
+values by default are ignored, but if you change `childCrossAxisExtent` to be
+100 pixels, all of the children are going to be 100 pixels in the main axis.
+This layout can be used with `DynamicGridView.wrap`.
 
 ## Getting started
 
-TODO: List prerequisites and provide or point to information on how to
+TODO(DavBot02): List prerequisites and provide or point to information on how to
 start using the package.
 
 ## Usage
 
-TODO: Include short and useful examples for package users. Add longer examples
-to `/example` folder.
+In this package, we use `DynamicGridViews` that are a class that is inherited
+from the normal `GridView`. You can use this `DynamicGridView` with
+constructors that have the specific `SliverGridDelegate` like
+`DynamicGridView.wrap` or if you want to have a more efficient option you can
+still use `DynamicGridView.builder` that works the same as `GridView.builder`.
+
+### Wrap
+
+The following are simple examples of how to use `DynamicGridView.wrap` and
+`DynamicGridView.builder` with the `SliverGridDelegateWithWrapping` delegate.
+
+<?code-excerpt "dynamic_grid_view_wrap.dart" (Example)?>
+```dart
+final List<Widget> children = List.generate(
+   250,
+   (index) => Container(
+     height: index.isEven ? 100 : 50,
+     width: index.isEven ? 95 : 180,
+     color: index.isEven ? Colors.red : Colors.blue,
+     child: Center(child: Text('Item $index')),
+   ),
+ );
+
+DynamicGridView.wrap(
+     mainAxisSpacing: 10,
+     crossAxisSpacing: 20,
+     children: children,
+);
+```
+
+Here is the result of the code:
+
+<video src="assets/simple_wrap_demo.mov">
+
+<?code-excerpt "dynamic_grid_view_builder.dart (Example)"?>
+```dart
+DynamicGridView.builder(
+     gridDelegate: const SliverGridDelegateWithWrapping(
+        mainAxisSpacing: 20,
+        childMainAxisExtent: 250,
+        childCrossAxisExtent: 50,
+     ),
+     itemBuilder: (BuildContext context, int index) {
+       return Container(
+         height: 200,
+         color: index.isEven ? Colors.amber : Colors.blue,
+         child: Center(
+           child: Text('$index'),
+         ),
+       );
+     },
+   ),
+```
+
+Here is the result of the code:
+
+<video src="assets/wrap_demo_with_one_fixed_axis.mov">
+
+### Staggered
+
+Using the Staggered layout is simple. It can be used with the
+`DynamicGridView.stagger` and you can still use the delegates from `GridView`
+like `SliverGridDelegateWithMaxCrossAxisExtent` and
+`SliverGridDelegateWithFixedCrossAxisCount`.
+
+<?code-excerpt "dynamic_grid_view_stagger.dart (Example)"?>
+```dart
+DynamicGridView.stagger(
+      crossAxisCount: 4,
+      mainAxisSpacing: 10.0,
+      crossAxisSpacing: 10.0,
+      children: children,
+);
+```
+
+<!-- TODO(snat-s): Add a video of DynamicGrid.stagger -->
```

Comment:
Maybe place all the TODOs in comments like this so they stand out more and don't accidentally render in the real readme.